### PR TITLE
created and added login api

### LINKED ISSUE #5: Add and test login API
# About
Adding and testing login API

# Why
The addition of a login API is essential for user authentication, enabling users to securely access protected resources and features within the application. This functionality enhances the overall security and user experience of the application.

# How
By creating new login function inside user.controller.js  

# Note
Thoroughly test the API to ensure its functionality and security. Cover various scenarios, including valid and invalid login.

# Reference

### PR DESCRIPTION
**Pull request by sujanshresstha**
# Changes
Created loginUser function inside user.controller.js

# How to test
- Test using postman
Screenshot 2023-12-02 at 15 39 18

# Reference
Screenshot 2023-12-02 at 15 40 06

# Issue
closed #5

### CODE REVIEW

File: snap-minds-api/src/controllers/user.controller.js [hunk 1]
```diff
@@ -79,34 +78,37 @@ const registerUser = asyncHandler( async (req, res) => {
...
+  }
 
-    const isMatch = await bcrypt.compare(password, user.password)
-    if (!isMatch) {
-      throw new ApiError(400, "Invalid credentials. ")
-    }
+  const isPasswordMatch = await userEmail.isPasswordCorrect(password);
+  if (!isPasswordMatch) {
+      throw new ApiError(409, "Invalid credentials.")
+  }
```
Comment (KHashimoto3) [inline snap-minds-api/src/controllers/user.controller.js]: 401 Unauthorized　にした方が良いです。

**Review by KHashimoto3 — CHANGES_REQUESTED** [review]
ステータスコードの変更をお願いします。

File: snap-minds-api/src/controllers/user.controller.js (hunk 1 above)
Comment (sujanshresstha) [reply snap-minds-api/src/controllers/user.controller.js]: Changed status code from 409 to 401 & removed password from the response.

**Review by KHashimoto3 — APPROVED** [review]
LGTM👍

**Pull request merged**